### PR TITLE
[DOCS-314] Add sprint 2 pentest contributors

### PR DESCRIPTION
**Pull request by mjang-cobalt**
## Changelog

Sprint 2 not yet complete. Including 3 approved articles.

### Updated

| Page | Deploy Preview | Comment |
| ----- | ----- | ----- |
| Best Practices | [[Link]](https://deploy-preview-210--cobalt-docs.netlify.app/bestpractices/) | 3 articles, so far |

## Preview This Change

To see how this change looks in production, scroll down to **Deploy Preview**. Select the link that looks like `https://deploy-preview-<num>--cobalt-docs.netlify.app/`

## Variables

Help us support a “Write once, publish everywhere” single source of truth. If you see a line that looks like:

`{{ % asset-categories % }}`

You’ve found a [shortcode](https://gohugo.io/content-management/shortcodes/) that we include in multiple documents.

You’ll find the content of the shortcode in the following directory:

https://github.com/cobalthq/cobalt-product-public-docs/tree/main/layouts/shortcodes

That shortcode has the same base name as what you see in the PR, such as `asset-categories.html`.

### CODE REVIEW

**Comment by mjang-cobalt** [comment]
@ana-dashuk-cobalt any idea why there's a gap between lines 3 and 4 of the table?
Screen Shot 2022-10-18 at 11 47 38 AM

**Comment by ana-dashuk-cobalt** [comment]
> @ana-dashuk-cobalt any idea why there's a gap between lines 3 and 4 of the table? Screen Shot 2022-10-18 at 11 47 38 AM

@mjang-cobalt The same thing happened last time. I tested this several times. The table doesn't break when I set up most (or all) links in shortcodes in HTML format.

Example:

```
<a href="https://harshbothra.tech/" target="_blank">Harsh Bothra</a>
```

instead of

```
[Harsh Bothra](https://harshbothra.tech/)
```

When we use shortcodes inside other elements, they may break the element. The same thing happens with Note callouts.

**Comment by mjang-cobalt** [comment]
Fixed with some copy/paste. (I moved some lines around, and the extra space always followed one of the lines) I'm also thinking, it's better to include the newest content atop the table.

Screen Shot 2022-10-18 at 12 50 14 PM

File: content/en/BestPractices/_index.md [hunk 1]
```diff
@@ -25,9 +25,11 @@ until we have enough articles to actually organize. -->
...
 
-| Article                                                              | Release Date | Author                |
-|----------------------------------------------------------------------|--------------|-----------------------|
-| [Validate User Input](./input-validation)                      | 2022-08-19   | {{% payloadartist %}} |
-| [Prevent Security Misconfiguration](./prevent-security-misconfig)              | 2022-08-19   | {{% shashank %}}      |
+| Article                                                               | Release Date | Author                |
+|-----------------------------------------------------------------------|--------------|-----------------------|
+| [Prevent Insecure Design in SQL](./secure-design)                     | 2022-10-18   | {{% shashank %}}      |
+| [Protect Your APIs With Rate Limiting](./api-rate-limiting)           | 2022-10-18   | {{% payloadartist %}} |
+| [Prevent Server-Side Template Injections](./prevent-ssti)             | 2022-10-18   | {{% payloadartist %}} |
```
Comment (ana-dashuk-cobalt) [inline content/en/BestPractices/_index.md:32]:
```suggestion
| [Prevent Server-Side Template Injections](./prevent-ssti)             | 2022-10-18   | {{% unstabl3 %}} |
```

**Comment by ana-dashuk-cobalt** [comment]
> Fixed with some copy/paste. (I moved some lines around, and the extra space always followed one of the lines) I'm also thinking, it's better to include the newest content atop the table.
> 
> Screen Shot 2022-10-18 at 12 50 14 PM

@mjang-cobalt The table looks great, but we're missing Shubham from the list.

**Review by ana-dashuk-cobalt — COMMENTED** [review]
@mjang-cobalt The table looks broken again.

Should we go ahead and add HTML to shortcodes to avoid this problem in the future?

**Comment by mjang-cobalt** [comment]
> Should we go ahead and add HTML to shortcodes to avoid this problem in the future?

I'm feeling dense. How would this help?

**Comment by mjang-cobalt** [comment]
I have a hacky fix:

Screen Shot 2022-10-19 at 10 09 16 AM

**Review by ana-dashuk-cobalt — APPROVED** [review]
@mjang-cobalt Approving because there is no better solution at the moment. I tried adding HTML instead of Markdown to shortcodes but it didn't work (last time it did). Since it's not a stable solution, we'll stick to the workaround that you suggested.

Or I would suggest adding Markdown directly to the table, without shortcodes. Essentially these links will only live on a single page, so it's easy to copy them from one row to another.

File: content/en/BestPractices/_index.md [hunk 2]
```diff
@@ -25,9 +25,18 @@ until we have enough articles to actually organize. -->
...
-| [Validate User Input](./input-validation)                      | 2022-08-19   | {{% payloadartist %}} |
-| [Prevent Security Misconfiguration](./prevent-security-misconfig)              | 2022-08-19   | {{% shashank %}}      |
-| [Protect Against Server-Side Request Forgery](./protect-against-ssrf) | 2022-08-19   | {{% harsh-bothra %}}  |
+### October 20, 2022
+
+| Article                                                               | Release Date | Author                |
+|-----------------------------------------------------------------------|--------------|-----------------------|
+| [Prevent Insecure Design in SQL](./secure-design)                     | 2022-10-18   | {{% shashank %}}      |
+| [Protect Your APIs With Rate Limiting](./api-rate-limiting)           | 2022-10-18   | {{% payloadartist %}} |
+| [Prevent Server-Side Template Injections](./prevent-ssti)             | 2022-10-18   | {{% unstabl3 %}}      |
```
Comment (ana-dashuk-cobalt) [inline content/en/BestPractices/_index.md:34]:
```suggestion
| [Prevent Insecure Design in SQL](./secure-design)                     | 2022-10-20   | {{% shashank %}}      |
| [Protect Your APIs With Rate Limiting](./api-rate-limiting)           | 2022-10-20   | {{% payloadartist %}} |
| [Prevent Server-Side Template Injections](./prevent-ssti)             | 2022-10-20   | {{% unstabl3 %}}      |
```

Comment (ana-dashuk-cobalt) [reply content/en/BestPractices/_index.md:34]: @mjang-cobalt

**Comment by mjang-cobalt** [comment]
> Or I would suggest adding Markdown directly to the table, without shortcodes. Essentially these links will only live on a single page, so it's easy to copy them from one row to another.

You may be right. 

I originally intended to include a sentence about each author in their articles, something like, "You can [read|learn] more about [authorname]'s work in [shortcode link]." But that doesn't work with current shortcodes. 

Maybe something simple, like include "Written by [shortcode author]." atop each page?

**Comment by ana-dashuk-cobalt** [comment]
> Maybe something simple, like include "Written by [shortcode author]." atop each page?

I like this idea! @mjang-cobalt

File: content/en/BestPractices/_index.md (hunk 2 above)
Comment (mjang-cobalt) [inline content/en/BestPractices/_index.md:34]:
```suggestion
### October 21, 2022

| Article                                                               | Release Date | Author                |
|-----------------------------------------------------------------------|--------------|-----------------------|
| [Prevent Insecure Design in SQL](./secure-design)                     | 2022-10-21   | {{% shashank %}}      |
| [Protect Your APIs With Rate Limiting](./api-rate-limiting)           | 2022-10-21   | {{% payloadartist %}} |
| [Prevent Server-Side Template Injections](./prevent-ssti)             | 2022-10-21   | {{% unstabl3 %}}      |
```

**Pull request merged**